### PR TITLE
FIX PYBIND11_SOFA_VERSION

### LINKED ISSUE #235: fail to make
Hi, Hugo.
I have been compiling SP3 using out of tree build, despite some Eigen warning,cmake was perfectly fine.
However,got some problem like this when execute cmake --build $SP3_build :

`code/SofaPython3-master/Plugin/src/SofaPython3/initModule.cpp:21:0:
/home/SENSETIME/xulixin2/code/SofaPython3-master/Plugin/src/SofaPython3/config.h:48:5: 
error: user-defined literal in preprocessor expression
 #if PYBIND11_SOFA_VERSION >= 20801
     ^
In file included from /usr/local/include/pybind11/cast.h:13:0,
                 from /usr/local/include/pybind11/attr.h:13,
                 from /usr/local/include/pybind11/detail/class.h:12,
                 from /usr/local/include/pybind11/pybind11.h:13,
                 from /home/SENSETIME/xulixin2/code/SofaPython3-master/Plugin/src/SofaPython3/config.h:24,
                 from /home/SENSETIME/xulixin2/code/SofaPython3-master/Plugin/src/SofaPython3/initModule.cpp:21:
/home/SENSETIME/xulixin2/code/SofaPython3-master/Plugin/src/SofaPython3/config.h:45:7: error:
 floating constant in preprocessor expression
     + PYBIND11_VERSION_PATCH)
       ^
/home/SENSETIME/xulixin2/code/SofaPython3-master/Plugin/src/SofaPython3/config.h:48:5: note: in expansion of macro ‘PYBIND11_SOFA_VERSION’
 #if PYBIND11_SOFA_VERSION >= 20801
`
I don't know how to deal with it, appreciate it if you could give some advice.

### PR DESCRIPTION
**Pull request by marques-bruno**
fixes #235 

Building SofaPython3 with the master version of pybind11 (or any "non-release" version with a patch tag in PYBIND11_VERSION_PATCH) will cause a compile error, because the PYBIND11_SOFA_VERSION macro definition expects a number-only macro.

There are two solutions to this issue:
1. is the one I'm proposing in this PR: removing the patch number from PYBIND11_SOFA_VERSION. I don't think it is necessary in SOFA to know the patch version of pybind11, but I might be wrong...

2. updating the documentation for SofaPython3, to explicitly tell people to either install pybind11 from their system package manager, or tell them to checkout a release version of pybind11 if one wants to build it from its source repository.

### CODE REVIEW

File: Plugin/src/SofaPython3/config.h [hunk 1]
```diff
@@ -41,8 +41,7 @@
 
 // define own pybind version macro, much easier to test/read
 #define PYBIND11_SOFA_VERSION (PYBIND11_VERSION_MAJOR * 10000 \
-    + PYBIND11_VERSION_MINOR * 100 \
-    + PYBIND11_VERSION_PATCH)
+    + PYBIND11_VERSION_MINOR * 100)
 // pybind11 already bind the attributeError starting from 2.8.1 version.
 // so if the version is >= 2.8.1, macro does nothing, otherwise bind attribute_error
 #if PYBIND11_SOFA_VERSION >= 20801
```
Comment (guparan) [inline Plugin/src/SofaPython3/config.h:47]: Actually just looking at this code makes me think that the patch version is useful.

Comment (guparan) [inline Plugin/src/SofaPython3/config.h:47]: Actually just looking at this code makes me think that the patch version is useful.

Comment (guparan) [reply Plugin/src/SofaPython3/config.h:47]: Another solution could be to use CMake generate to set `PYBIND11_SOFA_VERSION`. It implies to change this config.h into config.h.in
I can do it if you want.

Comment (marques-bruno) [reply Plugin/src/SofaPython3/config.h]: Much better in my opinion. I don't know how I missed the comment, right between the macro declaration and the version condition check....

I configured the macro with cmake in the last commit

File: Plugin/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 project(Plugin VERSION 1.0)
 
 set(HEADER_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/src/SofaPython3/config.h
+    ${CMAKE_CURRENT_SOURCE_DIR}/src/SofaPython3/config.h.in
```
Comment (guparan) [inline Plugin/CMakeLists.txt:4]: config.h.in will be automatically handled by the sofa_create_component_in_package_with_targets macro at the end of this file

**Pull request merged**